pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1576,6 +1593,50 @@ boolean isUpdateable() throws SQLException {
     return updateable;
   }
 
+  /**
+   * Method to turn on/off adaptive fetch for ResultSet.
+   *
+   * @param adaptiveFetch desired state of adaptive fetch.
+   * @throws SQLException exception returned if ResultSet is closed
+   */
+  public void setAdaptiveFetch(boolean adaptiveFetch) throws SQLException {
+    checkClosed();
+    updateQueryInsideAdaptiveFetchMonitoring(adaptiveFetch);
+    this.adaptiveFetch = adaptiveFetch;
+  }
+
+  /**
+   * Method to update adaptive fetch monitoring during changing state of adaptive fetch inside
+   * ResultSet. Update inside AdaptiveFetchMonitoring is required to collect data about max result
+   * row length for that query to compute adaptive fetch size.
+   *
+   * @param newAdaptiveFetch new state of adaptive fetch
+   */
+  private void updateQueryInsideAdaptiveFetchMonitoring(boolean newAdaptiveFetch) {
+    boolean localAdaptiveFetch = this.adaptiveFetch;
```

Comment:
why do we introduce this localAdaptifFetch variable here

Reply:
This variable was redundant, because was used before, when implementation of adaptive fetch had something like "global adaptive fetch" and "local adaptive fetch", then it was helpful, but as concept of two levels of adaptive fetch wasn't intuitive, it was removed and that variable was forgotten.